data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/buffer/InMemoryBufferFactory.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.buffer;
+
+public class InMemoryBufferFactory implements BufferFactory{
+    @Override
+    public Buffer getBuffer() {
+        return new InMemoryBuffer();
```

Comment:
No test coverage here. Is this going to be covered in another PR?

Reply:
I have added the test coverage.